service/src/chain_spec.rs

```diff
@@ -47,24 +50,27 @@ pub struct Extensions {
 	pub bad_blocks: sc_client::BadBlocks<polkadot_primitives::Block>,
 }
 
-/// The `ChainSpec`.
-///
-/// We use the same `ChainSpec` type for Polkadot and Kusama. As Kusama
-/// is only loaded from a file, the `GenesisConfig` type is not used.
-pub type ChainSpec = service::ChainSpec<
+/// The `ChainSpec parametrised for polkadot runtime`.
+pub type PolkadotChainSpec = service::GenericChainSpec<
 	polkadot::GenesisConfig,
 	Extensions,
 >;
 
-pub fn kusama_config() -> Result<ChainSpec, String> {
-	ChainSpec::from_json_bytes(&include_bytes!("../res/kusama.json")[..])
+/// The `ChainSpec parametrised for kusama runtime`.
+pub type KusamaChainSpec = service::GenericChainSpec<
+	kusama::GenesisConfig,
+	Extensions,
+>;
+
+pub fn kusama_config() -> Result<KusamaChainSpec, String> {
+	KusamaChainSpec::from_json_bytes(&include_bytes!("../res/kusama.json")[..])
 }
 
-pub fn westend_config() -> Result<ChainSpec, String> {
-	ChainSpec::from_json_bytes(&include_bytes!("../res/westend.json")[..])
+pub fn westend_config() -> Result<KusamaChainSpec, String> {
+	KusamaChainSpec::from_json_bytes(&include_bytes!("../res/westend.json")[..])
```

Comment:
```suggestion
	PolkadotChainSpec::from_json_bytes(&include_bytes!("../res/westend.json")[..])
```

Conceptual `Westend` is based on `Polkadot`.

Reply:
Is it based on the polkadot native runtime though?

Reply:
Yes.